lib/ts/recipe/thirdpartypasswordless/recipe.tsx

```diff
@@ -56,61 +62,48 @@ export default class ThirdPartyPasswordless extends AuthRecipe<
 
     thirdPartyRecipe: ThirdParty | undefined;
 
-    recipeImpl: TPPWlessRecipeInterface;
+    recipeImpl: WebJSRecipeInterface;
 
     constructor(
-        config: Config,
+        config: NormalisedConfigWithAppInfoAndRecipeID<NormalisedConfig>,
         recipes: {
             thirdPartyInstance: ThirdParty | undefined;
             passwordlessInstance: Passwordless | undefined;
         }
     ) {
-        super(normaliseThirdPartyPasswordlessConfig(config));
-
-        {
-            const builder = new OverrideableBuilder(
-                RecipeImplementation({
-                    appInfo: this.config.appInfo,
-                    recipeId: this.config.recipeId,
-                    onHandleEvent: this.config.onHandleEvent,
-                    preAPIHook: this.config.preAPIHook,
-                    postAPIHook: this.config.postAPIHook,
-                })
-            );
-            this.recipeImpl = builder.override(this.config.override.functions).build();
-        }
+        super(config);
+
+        this.recipeImpl = ThirdpartyPasswordlessWebJS;
```

Comment:
How is this working? Cause the type of the recipe != recipe implementation, and ThirdpartyPasswordlessWebJS is a recipe, whereas functions like `getPasswordlessImpl` expect a recipe implementation?

Reply:
probably because recipeImplementation and recipe have the same interface, and recipe wrapper calling the recipeImplementation underneath